core/consensus/grandpa/environment.hpp

```diff
@@ -6,54 +6,75 @@
 #ifndef KAGOME_CORE_CONSENSUS_GRANDPA_ENVIRONMENT_HPP
 #define KAGOME_CORE_CONSENSUS_GRANDPA_ENVIRONMENT_HPP
 
-#include "consensus/grandpa/round.hpp"
+#include "consensus/grandpa/chain.hpp"
+#include "consensus/grandpa/common.hpp"
+#include "consensus/grandpa/completed_round.hpp"
 #include "consensus/grandpa/structs.hpp"
 
 namespace kagome::consensus::grandpa {
 
   /// Necessary environment for a voter.
   ///
   /// This encapsulates the database and networking layers of the chain.
-  struct Environment {
-    virtual ~Environment() = default;
-
-    /// Produce data necessary to start a round of voting.
-    virtual RoundData roundData(RoundNumber round) = 0;
+  struct Environment : public Chain {
+    ~Environment() override = default;
 
+    // TODO(kamilsa): PRE-335 move timer to environment
     /// Return a timer that will be used to delay the broadcast of a commit
     /// message. This delay should not be static to minimize the amount of
     /// commit messages that are sent (e.g. random value in [0, 1] seconds).
-    virtual Timer roundCommitTimer() = 0;
+    // virtual Timer roundCommitTimer() = 0;
 
     /// Note that we've done a primary proposal in the given round.
-    virtual outcome::result<void> proposed(RoundNumber round,
-                                           PrimaryPropose propose) = 0;
+    /**
+     * Triggered when current peer appeared in round \param round and having
+     * \param set_id has \param propose ready to be gossiped.
+     */
+    virtual outcome::result<void> proposed(
```

Comment:
Might it be better to remove such methods to onProposed, onPrecommited, etc. ? I think it looks a bit more clear.